Core/MIPS/x86/X64IRCompALU.cpp

```diff
@@ -285,11 +286,75 @@ void X64JitBackend::CompIR_Shift(IRInst inst) {
 	case IROp::Shr:
 	case IROp::Sar:
 	case IROp::Ror:
+		CompIR_Generic(inst);
+		break;
+
 	case IROp::ShlImm:
+		// Shouldn't happen, but let's be safe of any passes that modify the ops.
+		if (inst.src2 >= 32) {
+			regs_.SetGPRImm(inst.dest, 0);
+		} else if (inst.src2 == 0) {
+			if (inst.dest != inst.src1) {
+				regs_.Map(inst);
+				MOV(32, regs_.R(inst.dest), regs_.R(inst.src1));
+			}
+		} else {
+			regs_.Map(inst);
+			if (inst.dest != inst.src1)
+				MOV(32, regs_.R(inst.dest), regs_.R(inst.src1));
+			SHL(32, regs_.R(inst.dest), Imm8(inst.src2));
+		}
+		break;
+
 	case IROp::ShrImm:
+		// Shouldn't happen, but let's be safe of any passes that modify the ops.
+		if (inst.src2 >= 32) {
+			regs_.SetGPRImm(inst.dest, 0);
+		} else if (inst.src2 == 0) {
+			if (inst.dest != inst.src1) {
+				regs_.Map(inst);
+				MOV(32, regs_.R(inst.dest), regs_.R(inst.src1));
+			}
```

Comment:
Ideally I guess previous passes should have taken care of this .. but maybe best not to assume.

Reply:
Yeah, I was running into some weird cases in RISC-V and some of these things are or might be HINTs there (i.e. certain instructions with args that are pointless may have "meaning" in the future.)  I don't think it was necessarily often...

-[Unknown]